### PR TITLE
Map State.STARTING as "starting" at fleet checkin to avoid presenting the agent as unhealthy at startup

### LINKED ISSUE #2272: [8.6.2]: Ubuntu agent gets unhealthy on installation temporarily.
**Similar issue under:** https://github.com/elastic/elastic-agent/issues/2145
 
**Kibana version:** 8.6.2 BC2 Kibana cloud environment

**Host OS:** AWS Ubuntu 22

**Build details:**
VERSION: 8.6.2 BC2 
BUILD: 59020
COMMIT: aabd37e8a9069bed4f63b8c457a6829f4581ab97
Artifact Link: https://staging.elastic.co/8.6.2-ca679c34/downloads/beats/elastic-agent/elastic-agent-8.6.2-linux-x86_64.tar.gz

**Preconditions:**
1. 8.6.2 BC2 Kibana cloud environment should be available.
2. Agent Policy should be available with System integration.
3. Another agent policy with System and Endpoint defend integration should be available.

**Steps to reproduce:**
1. Install ubuntu agents using both the policies.
2. Observe agents are installed Unhealthy under Agents tab.
3. Observe agent remain unhealthy for quite a long time and then itself gets Healthy.

**Note:**
- This issue is reproducible multiple times on different builds and fresh VMs.

**Screenshots:**

**Logs:**
**When installed with only System integration:**
[[System]elastic-agent-diagnostics-2023-02-14T17-49-25Z-00.zip](https://github.com/elastic/elastic-agent/files/10740646/System.elastic-agent-diagnostics-2023-02-14T17-49-25Z-00.zip)

**When installed with both System and Endpoint defend integrations:**
[[Endpoint]elastic-agent-diagnostics-2023-02-14T17-45-11Z-00.zip](https://github.com/elastic/elastic-agent/files/10740648/Endpoint.elastic-agent-diagnostics-2023-02-14T17-45-11Z-00.zip)

**Debug level logs:**
[[Debug]elastic-agent-diagnostics-2023-02-15T08-08-24Z-00.zip](https://github.com/elastic/elastic-agent/files/10740744/Debug.elastic-agent-diagnostics-2023-02-15T08-08-24Z-00.zip)

**Expected Result:**
 Agents should remain healthy on installation.

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?

This PR maps the State.STARTING into "starting" string sent to fleet during checkin.
This will prevent a starting agent to be flagged as unhealthy in fleet UI until the next checkin.

## Why is it important?

This should avoid confusion in the users when they see a starting agent with unhealthy state just after enrolling only to become healthy at next checkin

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)

## Author's Checklist

- [ ]

## Related issues

- Closes #2272 

## Screenshots
We need to add an additional log call at checkin time to see what's going on:

```
diff --git a/internal/pkg/agent/application/gateway/fleet/fleet_gateway.go b/internal/pkg/agent/application/gateway/fleet/fleet_gateway.go
index 9d9024ef5..748bf73f4 100644
--- a/internal/pkg/agent/application/gateway/fleet/fleet_gateway.go
+++ b/internal/pkg/agent/application/gateway/fleet/fleet_gateway.go
@@ -325,6 +325,12 @@ func (f *fleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	components := f.convertToCheckinComponents(state.Components)
 
 	// checkin
+
+	f.log.Infow("Checking In", "state", state.State, "mappedState", agentStateToString(state.State), "state_message", state.Message,
+		"fleetState", state.FleetState, "fleetMessage", state.FleetMessage,
+		"components", components,
+	)
+
 	cmd := fleetapi.NewCheckinCmd(f.agentInfo, f.client)
 	req := &fleetapi.CheckinRequest{
 		AckToken:   ackToken,
```

(mappedState is the status string the agent is sending to the fleet server)

Here's the log for the agent without the fix:
image (2)

And this is with the fix:
image

### CODE REVIEW

**Comment by cmacknz** [comment]
The change here looks reasonable to me. Is there a test we can write to prevent this type of regression? A unit test that passes every possible state to `agentStateToString` and ensures the result makes sense would have caught this.

**Comment by pchila** [comment]
Adding a test is certainly possible but we're on the edge of "too simple to break" since we would be testing a simple switch statement right now. I will add a test anyways mostly for documentation purposes ;)

File: internal/pkg/agent/application/gateway/fleet/fleet_gateway.go [hunk 1]
```diff
@@ -26,8 +26,11 @@ import (
 // Max number of times an invalid API Key is checked
 const maxUnauthCounter int = 6
 
-// Const for decraded state or linter complains
-const degraded = "DEGRADED"
+// Consts for states at fleet checkin
+const fleetStateDegraded = "DEGRADED"
+const fleetStateOnline = "online"
+const fleetStateError = "error"
+const fleetStateStarting = "starting"
```
Comment (ycombinator) [inline internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:33]: Is there some reason we are using all-uppercase for the degraded state and all-lowercase for the other states?

**Review by ycombinator — APPROVED** [review]
Just left a question.  But the change LGTM.

File: internal/pkg/agent/application/gateway/fleet/fleet_gateway.go (hunk 1 above)
Comment (pchila) [reply internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:33]: I guess this came from a linter issue asking for a constant since "DEGRADED" was used in 2 places, while the other stare are all constant strings used just once in code...
I preferred to have the constants used as output of state in one place but I didn't change the values :sweat_smile: 
But it bugs also my OCD seeing different casing (and also the "screaming" constant values a bit... )

I would love a quick refactor having all the statuses defined as constants and lowercase but maybe not in this bugfix ;)

Comment (cmacknz) [reply internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:33]: Agreed this reads oddly.

The agent reports status values in upper case in logs and in the output of the `elastic-agent status` command. Users have written scripts to explicitly look for these strings with the upper case values in test automation, deployment automation, etc. We changed the status values to all lowercase originally in 8.6 and then had to restore the upper case versions before release because of all the tests we broke across different teams 🤷 

The the states reported to Fleet are usually in lowercase because this is what the the Fleet UI represents them as internally: https://github.com/elastic/kibana/blob/bb35166fc3b9fdf5d2ab8da9849db0aef253f958/x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_health.tsx#L62-L80. 

I believe the UI calls `.toLowerCase` on these before looking at them so we could change all of these to uppercase if we wanted to.
